### PR TITLE
feat(ingest): add stateful ingestion support to looker and lookml source

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py [hunk 1]
```diff
@@ -82,11 +91,27 @@
...
+from datahub.utilities.source_helpers import (
+    auto_stale_entity_removal,
+    auto_status_aspect,
+)
 
 logger = logging.getLogger(__name__)
 
 
-class LookerDashboardSourceConfig(LookerAPIConfig, LookerCommonConfig):
+class LookerStatefulIngestionConfig(StatefulStaleMetadataRemovalConfig):
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:102]: the docs are wrong here - you can use `StatefulStaleMetadataRemovalConfig` directly below, and delete `LookerStatefulIngestionConfig` entirely

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py [hunk 2]
```diff
@@ -167,23 +198,26 @@ class LookerDashboardSource(TestableSource):
...
+        super().__init__(config, ctx)
         self.source_config = config
         self.reporter = LookerDashboardSourceReport()
         self.looker_api: LookerAPI = LookerAPI(self.source_config)
         self.user_registry = LookerUserRegistry(self.looker_api)
         self.explore_registry = LookerExploreRegistry(self.looker_api, self.reporter)
         self.reporter._looker_explore_registry = self.explore_registry
         self.reporter._looker_api = self.looker_api
+
+        self.explores_to_fetch_set: Dict[Tuple[str, str], List[str]] = {}
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:219]: why was this necessary?

**Review by hsheth2 — COMMENTED** [review]
some minor things but overall looking good

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:219]: This is not necessary if every ingestion (pipeline) runs in separate process. However if two pipelines are running in same process, it is observed that `explores_to_fetch_set` from earlier pipeline persisted for next pipeline run, leading to incorrect results. Noticed this when adding tests for stateful ingestion.

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:219]: ah that's pretty subtle - we should watch out for those sorts of bugs in our other sources, since we definitely do the same thing elsewhere

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:102]: Oh, so is there no need to override `_entity_types` according to source ?

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:102]: That's right - I've also opened a PR to remove it elsewhere in the codebase https://github.com/datahub-project/datahub/pull/6454

**Review by hsheth2 — APPROVED** [review]
LGTM!
